packages/dev/core/src/Misc/sceneSerializer.ts

```diff
@@ -40,62 +43,75 @@ const SerializeMesh = (mesh: Mesh, serializationScene: any): any => {
     return serializationObject;
 };
 
-const FinalizeSingleMesh = (mesh: Mesh, serializationObject: any) => {
-    //only works if the mesh is already loaded
-    if (mesh.delayLoadState === Constants.DELAYLOADSTATE_LOADED || mesh.delayLoadState === Constants.DELAYLOADSTATE_NONE) {
-        const serializeMaterial = (material: Material) => {
-            serializationObject.materials = serializationObject.materials || [];
-            if (mesh.material && !serializationObject.materials.some((mat: Material) => mat.id === (<Material>mesh.material).id)) {
-                serializationObject.materials.push(material.serialize());
-            }
-        };
-
-        //serialize material
-        if (mesh.material && !mesh.material.doNotSerialize) {
-            if (mesh.material instanceof MultiMaterial) {
-                serializationObject.multiMaterials = serializationObject.multiMaterials || [];
-                if (!serializationObject.multiMaterials.some((mat: Material) => mat.id === (<Material>mesh.material).id)) {
-                    serializationObject.multiMaterials.push(mesh.material.serialize());
-                    for (const submaterial of mesh.material.subMaterials) {
-                        if (submaterial) {
-                            serializeMaterial(submaterial);
+const FinalizeSingleNode = (node: Node, serializationObject: any) => {
+    // Mesh, GroundMesh, etc...
+    if (node.getClassName().indexOf("Mesh") !== -1) {

```

Comment:
What about `InstancedMesh`?

`InstancedMesh` extends `AbstractMesh` and not `Mesh`, so `delayLoadState` (for eg), does not exist for `InstancedMesh`.